File: LLD/LLD-2/src/TicTakToeGame/src/main/java/org/example/controllers/GameController.java
package org.example.controllers;

import org.example.models.Game;
import org.example.models.GameStatus;
import org.example.models.Player;
import org.example.strategies.winningStrategies.WinningStrategy;

import java.util.List;

public class GameController {
    public Game startGame(int boardSize, List<Player> players, List<WinningStrategy> winningStrategies) throws Exception {
        return Game.getBuilder()
                        .setDimension(boardSize)
                        .setPlayers(players)
                        .setWinningStrategies(winningStrategies)
                        .build();

    }

    public void makeMoves(Game game)
    {
        game.makeMove(game.getBoard());
    }

    public void displayBoard(Game game)
    {
        game.printBoard();
    }

    public Player checkWinner(Game game)
    {
        return game.getWinner();
    }

    public void undo(Game game)
    {
        game.undo();
    }

    public GameStatus getGameStatus(Game game)
    {
        return game.getGameStatus();
    }
}

File: LLD3/TicTacToe/TicTacToe/Controllers/GameController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TicTacToe.Models;
using TicTacToe.Strategies.WinningStrategies;

namespace TicTacToe.Controllers
{
    internal class GameController
    {
        public Game StartGame(int dimension, List<Player> players, List<IWinningStrategy> winning)
        {

            return Game.GetBuilder()
                            .SetDimension(dimension)
                            .SetPlayers(players)
                            .SetWinningStretegies(winning)
                            .Build();
        }

        public void MakeMove(Game game)
        {
            game.MakeMove();
        }

        public void DisplayBoard(Game game)
        {
            game.DisplayBoard();
        }

        public Player GetWinner(Game game)
        {
            return game.Winner;
        }

        public void Undo(Game game)
        {
            game.Undo();
        }
        public GameStatus GetStatus(Game game)
        {
            return game.GameStatus;
        }
    }
}
